spaceship/lib/spaceship/tunes/tunes_client.rb

```diff
@@ -856,6 +863,161 @@ def release!(app_id, version)
+    end
+
+    def create_iap_family(app_id: nil, name: nil, product_id: nil, reference_name: nil, versions: [])
+      r = request(:get, "ra/apps/#{app_id}/iaps/family/template")
+      data = parse_response(r, 'data')
+
+      data['activeAddOns'][0]['productId'] = { value: product_id }
+      data['activeAddOns'][0]['referenceName'] = { value: reference_name }
+      data['name'] = { value: name }
+      data["details"]["value"] = versions
+
+      r = request(:post) do |req|
+        req.url "ra/apps/#{app_id}/iaps/family/"
+        req.body = data.to_json
+        req.headers['Content-Type'] = 'application/json'
+      end
+      handle_itc_response(r.body)
+    end
+
+    # returns pricing goal array
+    def iap_subscription_pricing_target(app_id: nil, purchase_id: nil, currency:, tier:)
+      r = request(:get, "ra/apps/#{app_id}/iaps/#{purchase_id}/pricing/equalize/#{currency}/#{tier}")
+      parse_response(r, 'data')
+    end
+
+    # Creates an In-App-Purchases
+    def create_iap!(app_id: nil, type: nil, versions: nil, reference_name: nil, product_id: nil, cleared_for_sale: true, review_notes: nil, review_screenshot: nil, pricing_intervals: nil, family_id: nil, subscription_duration: nil, subscription_free_trial: nil)
+      # Load IAP Template based on Type
+      type ||= "consumable"
+      r = request(:get, "ra/apps/#{app_id}/iaps/#{type}/template")
+      data = parse_response(r, 'data')
+
+      # Now fill in the values we have
+      # some values are nil, that's why there is a hash
+      data['familyId'] = family_id.to_s if family_id
+      data['productId'] = { value: product_id }
+      data['referenceName'] = { value: reference_name }
+      data['clearedForSale'] = { value: cleared_for_sale }
+
+      data['pricingDurationType'] = { value: subscription_duration } if subscription_duration
+      data['freeTrialDurationType'] = { value: subscription_free_trial } if subscription_free_trial
+
+      # pricing tier
+      if pricing_intervals
+        data['pricingIntervals'] = []
+        pricing_intervals.each do |interval|
+          data['pricingIntervals'] << {
+              value: {
+                  country: interval[:country] || "WW",
+                  tierStem: interval[:tier].to_s,
+                  priceTierEndDate: interval[:end_date],
+                  priceTierEffectiveDate: interval[:begin_date]
+                }
+          }
+        end
+      end
+
+      versions_array = []
+      versions.each do |k, v|
+        versions_array << {
+                  value: {
+                    description: { value: v[:description] },
+                    name: { value: v[:name] },
+                    localeCode: k.to_s
+                  }
+        }
+      end
+      data["versions"][0]["details"]["value"] = versions_array
+      data['versions'][0]["reviewNotes"] = { value: review_notes }
+
+      if review_screenshot
+        # Upload Screenshot:
+        upload_file = UploadFile.from_path review_screenshot
```

Comment:
duplicated with create. Consider moving functionality for both cases to `app_version` where we have all uses of `UploadFile`